cmd/all-in-one/Dockerfile

```diff
@@ -25,43 +31,27 @@ EXPOSE 14250
 # Web HTTP
 EXPOSE 16686
 
-COPY all-in-one-linux-$TARGETARCH /go/bin/all-in-one-linux
 COPY sampling_strategies.json /etc/jaeger/
 
 VOLUME ["/tmp"]
-ENTRYPOINT ["/go/bin/all-in-one-linux"]
-CMD ["--sampling.strategies-file=/etc/jaeger/sampling_strategies.json"]
-
-FROM $debug_image AS debug
-ARG TARGETARCH=amd64
-
-# Agent zipkin.thrift compact
-EXPOSE 5775/udp
 
-# Agent jaeger.thrift compact
-EXPOSE 6831/udp
-
-# Agent jaeger.thrift binary
-EXPOSE 6832/udp
+FROM common AS release
 
-# Agent config HTTP
-EXPOSE 5778
+ARG TARGETARCH
+COPY all-in-one-linux-${TARGETARCH} /go/bin/all-in-one-linux
 
-# Collector HTTP
-EXPOSE 14268
+ENTRYPOINT ["/go/bin/all-in-one-linux"]
+CMD ["--sampling.strategies-file=/etc/jaeger/sampling_strategies.json"]
 
-# Collector gRPC
-EXPOSE 14250
+FROM common AS debug
 
-# Web HTTP
-EXPOSE 16686
+ARG TARGETARCH
+COPY all-in-one-debug-linux-${TARGETARCH} /go/bin/all-in-one-linux
 
 # Delve
 EXPOSE 12345
 
-COPY all-in-one-debug-linux-$TARGETARCH /go/bin/all-in-one-linux
-COPY sampling_strategies.json /etc/jaeger/
-
-VOLUME ["/tmp"]
 ENTRYPOINT ["/go/bin/dlv", "exec", "/go/bin/all-in-one-linux", "--headless", "--listen=:12345", "--api-version=2", "--accept-multiclient", "--log", "--"]
 CMD ["--sampling.strategies-file=/etc/jaeger/sampling_strategies.json"]
+
+FROM ${TARGET}
```

Comment:
I am zero knowledge of docker buildx, curious what is the meaning of have FROM at the end of Dockerfile ?

Reply:
Yes. 

it is not new of buildx.

FROM could be from a stage name.

i learn this from istio Dockerfiles, which is used to switch default and distroless